### PR TITLE
Update from bitpay

### PR DESCRIPTION
**Pull request by spazzymoto**
I have attached the testing document of the completed upgrade which includes the following libraries:

- particl-bitcore-lib
- bitcore-mnemonic
- bitcore-wallet-client
- bitcore-wallet-service
- copay

[Particl Copay App Upgrade.pdf](https://github.com/particl/particl-bitcore-lib/files/2722942/Particl.Copay.App.Upgrade.pdf)

### CODE REVIEW

File: lib/block/merkleblock.js [hunk 1]
```diff
@@ -149,26 +151,61 @@ MerkleBlock.prototype.validMerkleTree = function validMerkleTree() {
...
  * @param {Number} - depth - Current height
  * @param {Number} - pos - Current position in the tree
  * @param {Object} - opts - Object with values that need to be mutated throughout the traversal
+ * @param {Boolean} - checkForTxs - if true return opts.txs else return the Merkle Hash
  * @param {Number} - opts.flagBitsUsed - Number of flag bits used, should start at 0
  * @param {Number} - opts.hashesUsed - Number of hashes used, should start at 0
- * @param {Array} - opts.txs - Will finish populated by transactions found during traversal
+ * @param {Array} - opts.txs - Will finish populated by transactions found during traversal that match the filter
  * @returns {Buffer|null} - Buffer containing the Merkle Hash for that height
+ * @returns {Array} - transactions found during traversal that match the filter
```
Comment (kewde) [inline lib/block/merkleblock.js:197]: Two different return types and completely different content?

Comment (spazzymoto) [reply lib/block/merkleblock.js:197]: It seems like this came from the original bitcore-lib. From reading the code it seems like 3 things could be returned a buffer, null or array.

I think maybe they split the @returns to shorten the line width, but i'm not sure. i shall make it 1 returns

File: test/address.js [hunk 2]
```diff
@@ -545,6 +551,11 @@ describe('Address', function() {
       address.toString().should.equal('2N7T3TAetJrSCruQ39aNrJvYLhG1LJosujf');
     });
 
+    it('can create an address from a set of public keys with a nested witness program', function() {
+      var address = Address.createMultisig(publics, 2, Networks.livenet, true);
+      address.toString().should.equal('3PpK1bBqUmPK3Q6QPSUK7BQSZ1DMWL6aes');
```
Comment (rynomster) [inline test/address.js:556]: is this going to work? Isn't networks.livenet a particl network?

Comment (spazzymoto) [reply test/address.js:556]: Probably not, i fixed the errors preventing the tests from running to completion. But i haven't gone through the 1000 odd tests that fail to fix them. That will take a bit more time

**Pull request merged**